src/PiWeb.Import.Sdk/Modules/ImportFormat/IImportGroup.cs

```diff
@@ -0,0 +1,56 @@
+﻿#region copyright
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+/* Carl Zeiss Industrielle Messtechnik GmbH        */
+/* Softwaresystem PiWeb                            */
+/* (c) Carl Zeiss 2024                             */
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#endregion
+
+namespace Zeiss.PiWeb.Import.Sdk.Modules.ImportFormat;
+
+#region usings
+
+using System.Collections.Generic;
+
+#endregion
+
+/// <summary>
+/// Represents a set of files that will be imported, discarded or deferred depending on <see cref="ImportAction"/>.
+/// </summary>
+public interface IImportGroup
+{
+    #region properties
+
+    /// <summary>
+    /// The primary file with which the grouping was started. This file has to be recognized by the format.
+    /// </summary>
+    IFileSource PrimaryFile { get; }
+
+    /// <summary>
+    /// The list of additional files recognized by the import format and added using <see cref="AddFile"/>.
+    /// The list does not include <see cref="PrimaryFile"/>. These files will not be offered to other import formats.
+    /// </summary>
+    ICollection<IFileSource> AdditionalFiles { get; }
+
+    /// <summary>
+    /// The list of files recognized by the import format including <see cref="PrimaryFile"/>.
+    /// These files will not be offered to other import formats.
+    /// </summary>
+    ICollection<IFileSource> AllFiles { get; }
+
+    /// <summary>
+    /// Adds the <paramref name="fileSource"/> to this <see cref="IImportGroup"/>. 
+    /// These files will not be offered to other import formats. 
+    /// </summary>
+    void AddFile(IFileSource fileSource);

```

Comment:
Don't we want to allow deletion of file sources (aside the primary file)?

Reply:
I skipped it, because it was not needed yet. But as it is not much work I added it now nevertheless.